components/com_users/tmpl/login/default_login.php

```diff
@@ -120,8 +122,25 @@ class="btn btn-secondary w-100 <?php echo $button['class'] ?? '' ?>"
         <a class="com-users-login__remind list-group-item" href="<?php echo Route::_('index.php?option=com_users&view=remind'); ?>">
             <?php echo Text::_('COM_USERS_LOGIN_REMIND'); ?>
         </a>
-        <?php if ($usersConfig->get('allowUserRegistration')) : ?>
-            <a class="com-users-login__register list-group-item" href="<?php echo Route::_('index.php?option=com_users&view=registration'); ?>">
+        <?php
+        if ($usersConfig->get('allowUserRegistration')) :
+            $regLinkMenuId = $this->params->get('customRegLinkMenu');
+            $regLink = 'index.php?option=com_users&view=registration';
+
+            if ($regLinkMenuId) {
+                $menu = Factory::getApplication()->getMenu();
+                $item = $menu->getItem($regLinkMenuId);
+
+                if ($item) {
+                    $regLink = 'index.php?Itemid=' . $regLinkMenuId;
+
+                    if ($item->language !== '*' && MultiLanguage::isEnabled()) {
+                        $regLink .= '&lang=' . $item->language;
+                    }
+                }
```

Comment:
```suggestion
                $regLink = 'index.php?Itemid=' . $regLinkMenuId;
```

Not sure why you need all the other stuff, if the link doesn't exists anymore that's then you should not fallback to the default registration because it might be unexpected to the site owner. Also language should be handled by the Router himself.

Or what do I miss here?

Reply:
this code is just replicated from the module `modules\mod_login\src\Helper\LoginHelper.php`

if it is wrong here then it is wrong there

Reply:
Has been fixed with https://github.com/joomla/joomla-cms/pull/45715/commits/442d590d73d34376fe1922e7d1139bc99be53234 .